docs/middle-component.md

```diff
@@ -56,3 +56,33 @@ from("sipmc:foo")
 from("sipmc:foo")
     .to("mock:output2");
 ```
+
+## Unit testing with sipmc
+
+Middle Component should be mocked within unit tests which test and cover Camel routes.
+Otherwise, running the test will run the route within Camel context without Middle component and test will fail running.
+
+When doing Camel testing, Middle component can be mocked by using Camel AdwiceWith inside your test. Following code and 
+methods within Advice can be used for mocking sipmc: 
+
+
+```java
```

Comment:
Please mention somewhere that this code snippet depends on extending CamelTestSupport